CMakeLists.txt

```diff
@@ -11,66 +11,52 @@ if(CMAKE_COMPILER_IS_GNUCXX OR CMAKE_CXX_COMPILER_ID MATCHES "Clang")
 endif()
 
 find_package(ament_cmake REQUIRED)
+find_package(builtin_interfaces REQUIRED)
 find_package(geometry_msgs REQUIRED)
+find_package(kdl_parser REQUIRED)
+find_package(orocos_kdl REQUIRED)
 find_package(rclcpp REQUIRED)
+find_package(rclcpp_components REQUIRED)
 find_package(sensor_msgs REQUIRED)
-find_package(orocos_kdl REQUIRED)
+find_package(std_msgs REQUIRED)
+find_package(tf2_ros REQUIRED)
 find_package(urdf REQUIRED)
 find_package(urdfdom_headers REQUIRED)
-find_package(kdl_parser REQUIRED)
-find_package(tf2_ros REQUIRED)
 
-include_directories(
-  include
-  ${geometry_msgs_INCLUDE_DIRS}
-  ${kdl_parser_INCLUDE_DIRS}
-  ${orocos_kdl_INCLUDE_DIRS}
-  ${urdf_INCLUDE_DIRS}
-  ${urdfdom_headers_INCLUDE_DIRS}
-  ${rclcpp_INCLUDE_DIRS}
-  ${sensor_msgs_INCLUDE_DIRS}
-  ${tf2_ros_INCLUDE_DIRS})
+include_directories(include)
 
 add_library(
-  ${PROJECT_NAME}_solver
+  ${PROJECT_NAME} SHARED
   src/robot_state_publisher.cpp)
-link_directories(${orocos_kdl_LIBRARY_DIRS})
-target_link_libraries(
-  ${PROJECT_NAME}_solver
-  ${geometry_msgs_LIBRARIES}
-  ${kdl_parser_LIBRARIES}
-  ${orocos_kdl_LIBRARIES}
-  ${rclcpp_LIBRARIES}
-  ${sensor_msgs_LIBRARIES}
-  ${tf2_ros_LIBRARIES}
-  ${urdf_LIBRARIES}
-  ${urdfdom_LIBRARIES})
+ament_target_dependencies(${PROJECT_NAME}
+  builtin_interfaces
+  geometry_msgs
+  kdl_parser
+  orocos_kdl
+  rclcpp
+  rclcpp_components
+  sensor_msgs
+  std_msgs
+  tf2_ros
+  urdf
+)
 
-add_library(
-  joint_state_listener
-  src/joint_state_listener.cpp)
-target_link_libraries(
-  joint_state_listener
-  ${PROJECT_NAME}_solver
-  ${orocos_kdl_LIBRARIES})
+rclcpp_components_register_nodes(${PROJECT_NAME}
+  "robot_state_publisher::RobotStatePublisher")
 
 add_executable(
+  ${PROJECT_NAME}_node
```

Comment:
You may be able to avoid this boilerplate if you use `rclcpp_components_register_node` (https://github.com/ros2/rclcpp/blob/master/rclcpp_components/cmake/rclcpp_components_register_node.cmake#L28), which will also create the executable for you.

Reply:
Yeah, good point, I'll switch to that.